File: src/main/java/com/ngolajr/ispmedia/entities/Album.java
package com.ngolajr.ispmedia.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.UUID;

//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
@Entity
@ToString
@Getter
@Setter
public class Album extends Conteudo{
    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Musica> musics=new ArrayList<>();
    //@ManyToOne(fetch = FetchType.EAGER ,cascade = CascadeType.ALL)
    @JsonBackReference(value = "artista-album")
    @ManyToOne
    private Artista artista;
    private Date dataLancamento;
    private double pontuacaoMedia;
    private int streams;
    public Album(){
        super();
        this.streams = 0;
    }
}

File: ISPMEDIA-ANGULAR/src/app/entities/Album.ts
import { Conteudo } from './Conteudo';
import { Musica } from './Musica';
import { Artista } from './Artista';
import { Critica } from './Critica';
import {Genero} from "./Genero";
import {User} from "./User";

export class Album extends Conteudo {
  musics: Musica[]|null=null;
  artista: Artista|null=null;
  criticas: Critica[]=[];
  pontuacaoMedia: number=0;
  dataLancamento:string="";
  streams : number=0;

  constructor(id: string, titulo: string, thumbNailUri: string, descricao: string, genero:Genero,editora:string,
              criador:User, musics: Musica[]|null, artista: Artista|null, criticas: Critica[], pontuacaoMedia: number,
              dataLancamento:string, streams:number) {
    super(id, titulo, thumbNailUri, descricao, genero, editora, criador);
    this.musics = musics;
    this.artista = artista;
    this.criticas = criticas;
    this.pontuacaoMedia = pontuacaoMedia;
    this.streams = streams;
    this.dataLancamento = dataLancamento;
  }
}
